server/src/bge/System.cpp

```diff
@@ -593,4 +593,83 @@ namespace bge {
             }
         }
     }
+
+    // ------------------------------------------------------------------------------------------------------------------------------------------------
+
+    SeasonEffectSystem::SeasonEffectSystem(
+        World* _gameWorld,
+		std::shared_ptr<ComponentManager<HealthComponent>> _healthCM,
+		std::shared_ptr<ComponentManager<VelocityComponent>> _velocityCM,
+        std::shared_ptr<ComponentManager<MovementRequestComponent>> _movementRequestCM, 
+        std::shared_ptr<ComponentManager<JumpInfoComponent>> _jumpInfoCM, 
+        std::shared_ptr<ComponentManager<SeasonAbilityStatusComponent>> _seasonAbilityStatusCM) {
+        
+        world = _gameWorld;
+        healthCM = _healthCM;
+        velocityCM = _velocityCM;
+        movementRequestCM = _movementRequestCM;
+        jumpInfoCM = _jumpInfoCM;
+        seasonAbilityStatusCM = _seasonAbilityStatusCM;
+        counter = 0;
+
+    }
+
+    void SeasonEffectSystem::update() {
+
+        // Change this to make each season longer or shorter
+        if (counter > 500) {
+            counter = 0;
+            world->currentSeason = (world->currentSeason+1)%4;
+            // std::printf("Current Season is %d\n", world->currentSeason);
+        }
+
+        if (world->currentSeason == SPRING_SEASON) {
+            for (Entity e : registeredEntities) {
+                HealthComponent& health = healthCM->lookup(e);
+                if (counter % 50 == 0) {
+                    health.healthPoint = std::min(100,health.healthPoint+5);
```

Comment:
[nit] I think it would be better for consistency to use PLAYER_MAX_HEALTH (or define a new constant if you want Spring to change the health cap)

Reply:
Oh yeah didn't realize we had that. Changed this.